mailbox/store/src/main/java/org/apache/james/mailbox/store/FlagsUpdateCalculator.java

```diff
@@ -33,6 +33,22 @@ public FlagsUpdateCalculator(Flags providedFlags, MessageManager.FlagsUpdateMode
         this.mode = mode;
     }
 
+    public Flags getProvidedFlags() {
+        return new Flags(providedFlags);
+    }
+
+    public MessageManager.FlagsUpdateMode getMode() {
+        return mode;
+    }
+
+    public boolean isFlagModified(Flags.Flag flag) {
+        if (mode == MessageManager.FlagsUpdateMode.ADD
```

Comment:
With REPLACE we should always set the flag to the value specified, hence it is always true